types/prosemirror-view/prosemirror-view-tests.ts

```diff
@@ -68,6 +68,25 @@ const res6_plugin = new state.Plugin({
         clipboardTextParser(text, $context, plain) {
             const isPlain: boolean = plain;
             return model.Slice.empty;
-        }
-    }
+        },
+    },
+});
+
+const view1 = new state.Plugin({
+    props: {
+        handleDOMEvents: {
+            click(view, event) {
+                event; // $ExpectType MouseEvent
+                return true;
+            },
+            blur(view, event) {
+                event; // $ExpectType FocusEvent
+                return true;
+            },
+            customxyz: (view, event) => {
```

Comment:
`event` intentionally defaults to any so that types pass correctly, and also it allows for typing the function with a custom event if desired in a much simpler way. 

```ts
handleDOMEvents: {
  // This would have been an error previously.
  custom: (view, event: MyCustomEvent) => {
	// ...
  }
}
```

Reply:
The TypeScript handbook [recommends](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#any) not using `any`:

> Don’t use `any` as a type unless you are in the process of migrating a JavaScript project to TypeScript. The compiler effectively treats `any` as “please turn off type checking for this thing”. It is similar to putting an `@ts-ignore` comment around every usage of the variable. This can be very helpful when you are first migrating a JavaScript project to TypeScript as you can set the type for stuff you haven’t migrated yet as any, but in a full TypeScript project you are disabling type checking for any parts of your program that use it.
> 
> In cases where you don’t know what type you want to accept, or when you want to accept anything because you will be blindly passing it through without interacting with it, you can use unknown.

I think we should follow that advice. To preserve backwards-compatibility, we should use `Event` rather than `unknown`.

When using custom events, one could add them to `DocumentEventMap` (using declaration merging) and achieve a similar effect but in a more type-safe way.

Reply:
I initially wanted to use `Event` which I tried first. But it errors since the intersection of the types have no overlap. `unknown` also has a similar error.

```
Type '{ click(this: Plugin<any, any>, view: EditorView<any>, event: MouseEvent): true; blur(this: Plugin<...>, view: EditorView<any>, event: FocusEvent): true; customxyz: (this: Plugin<...>, view: EditorView<...>, event: Event) => true; }' is not assignable to type 'HandleDOMEventsProp<Plugin<any, any>, any>'.
  Type '{ click(this: Plugin<any, any>, view: EditorView<any>, event: MouseEvent): true; blur(this: Plugin<...>, view: EditorView<any>, event: FocusEvent): true; customxyz: (this: Plugin<...>, view: EditorView<...>, event: Event) => true; }' is not assignable to type '{ [key: string]: (this: Plugin<any, any>, view: EditorView<any>, event: Event) => boolean; }'.
    Property 'click' is incompatible with index signature.
      Type '(this: Plugin<any, any>, view: EditorView<any>, event: MouseEvent) => true' is not assignable to type '(this: Plugin<any, any>, view: EditorView<any>, event: Event) => boolean'.
        Types of parameters 'event' and 'event' are incompatible.
          Type 'Event' is missing the following properties from type 'MouseEvent': altKey, button, buttons, clientX, and 20 more.
```

The reason I used `any` was because, if a user really wants to add a custom type they can use **Declaration Merging** or specifically add their type to the argument. I can add a comment, which outlines how to do it?

#### Error screenshot

![image](https://user-images.githubusercontent.com/1160934/95745975-f4af8e00-0c8d-11eb-83ba-44ec91afdd86.png)

Reply:
As an alternative, I could remove the intersection with `{ [key: string]: (this: ThisT, view: EditorView<S>, event: Event) => boolean; }`. That way if you want to add a custom type, you need to use declaration merging. I avoided this because I don't think it's a great experience for users of the types.

While I understand the reasoning behind not using `any` I think in this case, to enhance usability it any is the better choice. 

The file already has multiple uses of the `any` type so it's not a reduction in quality.

Reply:
Thanks for explaining the reasoning behind that choice! I think that both the current approach (using `any`) and the one without `{ [key: string]: (this: ThisT, view: EditorView<S>, event: Event) => boolean; }` have their pros and cons. You can leave it as-is or change it, I would accept it either way.